src/test/CMakeLists.txt

```diff
@@ -211,6 +211,10 @@ function(add_all_test_targets)
   endforeach()
 endfunction()
 
+if (BUILD_KERNEL_TEST)
+  add_subdirectory(kernel)
+endif()
+
```

Comment:
I'm not sure if this is the best approach. `-DBUILD_KERNEL_LIB=ON -DBUILD_KERNEL_TEST=ON` should imo build the tests even if `-DBUILD_TESTS=OFF`. I think an approach where we update `src/CMakeLists.txt` with the below makes more sense (quick sketch)?

```cmake
if(BUILD_KERNEL_LIB)
  add_subdirectory(kernel)
  if (BUILD_KERNEL_TEST)
    add_subdirectory(test/kernel)
  endif()
endif()
```